src/libraries/System.Text.Json/tests/JsonNode/JsonValueTests.cs

```diff
@@ -0,0 +1,291 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using System.Text.Json.Serialization;
+using Xunit;
+
+namespace System.Text.Json.Node.Tests
+{
+    public static class JsonValueTests
+    {
+        [Fact]
+        public static void CreateFromNull()
+        {
+            Assert.Null(JsonValue.Create((object)null));
+        }
+
+        [Fact]
+        public static void CreateFromNode_Fail()
+        {
+            Assert.Throws<ArgumentException>(() => JsonValue.Create(new JsonArray()));
+            Assert.Throws<ArgumentException>(() => JsonValue.Create(new JsonObject()));
+            Assert.Throws<ArgumentException>(() => JsonValue.Create(JsonValue.Create(42)));
+        }
+
+        [Fact]
+        public static void GetValue_Throws()
+        {
+            Assert.Throws<InvalidOperationException>(() => JsonNode.Parse("{}").GetValue<object>());
+            Assert.Throws<InvalidOperationException>(() => JsonNode.Parse("[]").GetValue<object>());
+        }
+
+        private class Polymorphic_Base { }
+        private class Polymorphic_Derived : Polymorphic_Base { }
+
+        [Fact]
+        public static void Polymorphic()
+        {
+            JsonValue value;
+
+            Polymorphic_Base baseClass = new Polymorphic_Derived();
+            value = JsonValue.Create(baseClass);
+            Assert.Same(baseClass, value.GetValue<Polymorphic_Derived>());
+
+            Polymorphic_Derived derivedClass = new Polymorphic_Derived();
+            value = JsonValue.Create(derivedClass);
+            Assert.Same(derivedClass, value.GetValue<Polymorphic_Base>());
+
+            Assert.Same(derivedClass, value.GetValue<object>());
+        }
+
+        [Fact]
+        public static void QuotedNumbers_Deserialize()
+        {
+            var options = new JsonSerializerOptions();
+            options.NumberHandling = JsonNumberHandling.AllowReadingFromString |
+                JsonNumberHandling.AllowNamedFloatingPointLiterals;
+
+            JsonNode node = JsonSerializer.Deserialize<JsonNode>("\"42\"", options);
+            Assert.IsAssignableFrom<JsonValue>(node);
+            Assert.Throws<InvalidOperationException>(() => node.GetValue<int>());
```

Comment:
Is it possible to have this work in the future?

Reply:
Yes if we decide we want to ease up on the `GetValue()` semantics and invoke the serializer. That would also require the `JsonSerializerOptions` to be passed in for things like quoted numbers.

We could also add a new method, like Json.NET's `JNode.ToObject()`, to deserialize.